fe/src/main/java/org/apache/doris/analysis/TimestampArithmeticExpr.java

```diff
@@ -120,65 +120,142 @@ private Type fixType() {
+                dateType = Type.DATETIME;
+            }
+            // The first child must return a timestamp or null.
+            if (!getChild(0).getType().isDateType() && !getChild(0).getType().isNull()) {
+                if (!dateType.isValid()) {
+                    throw new AnalysisException("Operand '" + getChild(0).toSql()
+                            + "' of timestamp arithmetic expression '" + toSql() + "' returns type '"
+                            + getChild(0).getType() + "'. Expected type 'TIMESTAMP/DATE/DATETIME'.");
+                }
+                castChild(dateType, 0);
             }
-            castChild(dateType, 0);
-        }
 
-        if (!getChild(1).getType().isScalarType()) {
-            throw new AnalysisException("must be a scalar type.");
-        }
+            if (!getChild(1).getType().isScalarType()) {
+                throw new AnalysisException("must be a scalar type.");
+            }
 
-        // The second child must be of type 'INT' or castable to it.
-        if (!getChild(1).getType().isScalarType(PrimitiveType.INT)) {
-            if (!ScalarType.canCastTo((ScalarType) getChild(1).getType(), Type.INT)) {
-                throw new AnalysisException("Operand '" + getChild(1).toSql()
-                        + "' of timestamp arithmetic expression '" + toSql() + "' returns type '"
-                        + getChild(1).getType() + "' which is incompatible with expected type 'INT'.");
+            // The second child must be of type 'INT' or castable to it.
+            if (!getChild(1).getType().isScalarType(PrimitiveType.INT)) {
+                if (!ScalarType.canCastTo((ScalarType) getChild(1).getType(), Type.INT)) {
+                    throw new AnalysisException("Operand '" + getChild(1).toSql()
+                            + "' of timestamp arithmetic expression '" + toSql() + "' returns type '"
+                            + getChild(1).getType() + "' which is incompatible with expected type 'INT'.");
+                }
+                castChild(Type.INT, 1);
             }
-            castChild(Type.INT, 1);
-        }
 
-        type = dateType;
-        opcode = getOpCode();
-        String funcOpName = String.format("%sS_%s",  timeUnit,
-                (op == ArithmeticExpr.Operator.ADD) ? "ADD" : "SUB");
-        // For the month interval, use the invisible special-case implementation.
-        // "ADD_MONTHS(t, m)" by definition is different from "t + INTERVAL m MONTHS".
-        // if (timeUnit == TimeUnit.MONTH) {
-        //     funcOpName += "_INTERVAL";
-        // }
+            type = dateType;
+            opcode = getOpCode();
+            funcOpName = String.format("%sS_%s", timeUnit,
+                    (op == ArithmeticExpr.Operator.ADD) ? "ADD" : "SUB");
+        }
 
         fn = getBuiltinFunction(analyzer, funcOpName.toLowerCase(),
                 collectChildReturnTypes(), Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
```

Comment:
`fn is {} name is {}` thie log should be remove or change to debug level.  There will be a lot of this log from doris 0.11

Reply:
OK